src/main/java/org/beanpod/switchboard/entity/StreamEntity.java

```diff
@@ -35,4 +36,8 @@
   @OneToOne(optional = false, fetch = FetchType.LAZY)
   @JoinColumn(name = "output_channel_id")
   private OutputChannelEntity outputChannel;
+
+  @Column(name = "mode")
+  @NotNull
+  private StreamModeEntity mode;
```

Comment:
Because this is either SRT or not SRT you might want to use a Boolean for the mode field

Reply:
I chose an Enum to allow for more felxibility if we want to add a streaming modes in the future.

Reply:
That’s one way of looking at the problem. I’m not convinced that Enums is the way to go.

Right now I’m only viewing the problem as us needing to implement the SRT protocol for streaming, but if we needed to extend our service to have compatibility with other streaming protocols I would more rely on some form of inheritance that creates a stream of a certain type allowing us to achieve a desired behaviour based on the stream object run type. If you have one stream class for all possible types of streams, and each streaming protocol has their own unique characteristics, then extension will increase our need for if-else logic, and most likely result in a bloated stream class with fields going unused depending on the stream protocol.

I’m mostly just thinking aloud and I’ve yet to gather all my thoughts. I might need to review our design goals. But, what I’m saying is that for now it may be sufficient to just think about the SRT streaming protocol unless our design must support multiple streaming protocols?

And, if we need to support multiple streaming protocols down the road, then take the time to evaluate the design tradeoffs of using an Enum field vs other design options.

Reply:
Not to force your hand out of enums this is a suggestion rather than a change request, but I just wanted to point out some issues you are likely to encounter if you go down this route. A Boolean on the other hand seems to fit better with what you are currently trying to accomplish, and is not as likely to cause you issues.

Reply:
I ended up changing it to `boolean isRendezvous` mode as you suggested. For reference, the Enum would previously be saved as its integer value, which I think could cause some problems if the Enum order is changed. In the future if we need to persist the enum, we should annonate the entity parameter with `@Enumerated`. For example, we could use `@Enumerated(EnumType.STRING)` to save the enum as a string.

Reply:
sounds good! to clarify the design goals at the moment, we only need to support SRT.